src/document/crdt/rga_tree_split.ts

```diff
@@ -825,17 +836,48 @@ export class RGATreeSplit<T extends RGATreeSplitValue> {
           createdAtMapByActor.set(actorID, node.getID().getCreatedAt());
         }
         removedNodeMap.set(node.getID().getAnnotatedString(), node);
+      } else if (!node.isRemoved()) {
+        nodesToKeep.push(node);
       }
     }
+    nodesToKeep.push(rightEdge);
 
     for (const node of nodesToDelete) {
       node.remove(editedAt);
-      this.treeByIndex.splayNode(node);
+      node.setWeight(0);
```

Comment:
Can we use `initWeight` here?
```suggestion
      node.initWeight();
```